Build failed on FreeBSD i386, versions 11.4, 12.2, 13.0, -CURRENT

Hi,

I've added RedisJSON to the FreeBSD ports tree and reported about that a couple of weeks ago.
Our build system has found an issue when it builds RedisJSON on i386 platform, doesn't matter what version of the operation system it is:
- FreeBSD 11.4 i386 - http://beefy10.nyi.freebsd.org/data/114i386-default/ca3f9a8e0cde/logs/redisjson-0.0.0.0.2021.08.10.01.log
- FreeBSD 12.2 i386 - http://beefy5.nyi.freebsd.org/data/122i386-default/ca3f9a8e0cde/logs/redisjson-0.0.0.0.2021.08.10.01.log
- FreeBSD 13.0 i386 - http://beefy15.nyi.freebsd.org/data/130i386-default/ca3f9a8e0cde/logs/redisjson-0.0.0.0.2021.08.10.01.log
- FreeBSD-CURRENT - http://beefy17.nyi.freebsd.org/data/main-i386-default/pf9c9fbc15f10_s159258afb5/logs/redisjson-0.0.0.0.2021.08.10.01.log

The build issue is identical on all versions and here it is:
error[E0308]: mismatched types                                                                                                        
   --> src/c_api.rs:139:29                                                                                                            
    |                                                                                                                                 
139 |             unsafe { *val = json.get_long() };                                                                                  
    |                             ^^^^^^^^^^^^^^^ expected `i32`, found `i64`                                                         
                                                                                                                                      
error: aborting due to previous error                                                                                                 
                                                                                                                                      
For more information about this error, try `rustc --explain E0308`.                                                                   
error: could not compile `redisjson`

Could you please take a look.
Thanks in advance.

--- a/src/c_api.rs
+++ b/src/c_api.rs
@@ -1,6 +1,6 @@
 use libc::size_t;
 use std::ffi::CString;
-use std::os::raw::{c_double, c_int, c_long};
+use std::os::raw::{c_double, c_int, c_longlong};
 use std::ptr::{null, null_mut};
 use std::{
     ffi::CStr,
@@ -135,7 +135,7 @@ pub fn json_api_get_json<M: Manager>(
 }
 
 #[allow(clippy::not_unsafe_ptr_arg_deref)]
-pub fn json_api_get_int<M: Manager>(_: M, json: *const c_void, val: *mut c_long) -> c_int {
+pub fn json_api_get_int<M: Manager>(_: M, json: *const c_void, val: *mut c_longlong) -> c_int {
     let json = unsafe { &*(json as *const M::V) };
     match json.get_type() {
         SelectValueType::Long => {
@@ -434,7 +434,7 @@ macro_rules! redis_json_module_export_shared_api {
         }
 
         #[no_mangle]
-        pub extern "C" fn JSONAPI_getInt(json: *const c_void, val: *mut c_long) -> c_int {
+        pub extern "C" fn JSONAPI_getInt(json: *const c_void, val: *mut c_longlong) -> c_int {
             $pre_command_function_expr(&get_llapi_ctx(), &Vec::new());
 
             let m = $get_manager_expr;
@@ -594,7 +594,7 @@ macro_rules! redis_json_module_export_shared_api {
             pub getAt: extern "C" fn(json: *const c_void, index: size_t) -> *const c_void,
             pub getLen: extern "C" fn(json: *const c_void, len: *mut size_t) -> c_int,
             pub getType: extern "C" fn(json: *const c_void) -> c_int,
-            pub getInt: extern "C" fn(json: *const c_void, val: *mut c_long) -> c_int,
+            pub getInt: extern "C" fn(json: *const c_void, val: *mut c_longlong) -> c_int,
             pub getDouble: extern "C" fn(json: *const c_void, val: *mut c_double) -> c_int,
             pub getBoolean: extern "C" fn(json: *const c_void, val: *mut c_int) -> c_int,
             pub getString: extern "C" fn(

--- a/src/lib.rs
+++ b/src/lib.rs
@@ -70,7 +70,7 @@ macro_rules! redis_json_module_create {(
 
         use redis_module::{redis_command, redis_module, RedisString};
         use std::marker::PhantomData;
-        use std::os::raw::{c_double, c_int, c_long};
+        use std::os::raw::{c_double, c_int, c_longlong};
         use redis_module::{raw as rawmod};
         use rawmod::ModuleOptions;
         use std::{

